integrations/dd-sdk-android-okhttp/src/main/kotlin/com/datadog/android/okhttp/DatadogInterceptor.kt

```diff
@@ -397,6 +417,48 @@ internal constructor(
 
     // endregion
 
+    // region Builder
+
+    /**
+     * A Builder for the [DatadogInterceptor].
+     */
+    open class Builder(tracedHosts: Map<String, Set<TracingHeaderType>>) :
+        BaseBuilder<DatadogInterceptor, Builder>(tracedHosts) {
+
+        private var rumResourceAttributesProvider: RumResourceAttributesProvider = NoOpRumResourceAttributesProvider()
+
+        override fun getThis(): Builder {
```

Comment:
`getThis`? why is it needed?

Reply:
`Generics` check the `Builder` and `BaseBuilder` class. It is just a patter for having a `BaseBuilder` using Java Generics. In any case this is `internal`

Reply:
> [!TIP]
> Maybe this method could be `protected`, otherwise it might be confusing to our customers